testing/dart/compositing_test.dart

```diff
@@ -359,6 +359,23 @@ void main() {
         oldLayer: oldLayer,
       );
     });
+    testNoSharing((SceneBuilder builder, EngineLayer oldLayer) {
+      return builder.pushImageFilter(
+        ImageFilter.blur(sigmaX: 10.0, sigmaY: 10.0),
+        oldLayer: oldLayer,
+      );
+    });
+    testNoSharing((SceneBuilder builder, EngineLayer oldLayer) {
+      return builder.pushImageFilter(
+        ImageFilter.matrix(Float64List.fromList(<double>[
+          1, 0, 0, 0,
+          0, 1, 0, 0,
+          0, 0, 1, 0,
+          0, 0, 0, 1,
+        ])),
+        oldLayer: oldLayer,
+      );
+    });
   });
```

Comment:
we should also test the actual pixels do what we expect

Reply:
I was following the trail of ColorFilter tests and implemented similar tests to what I found there. I agree that it would be good to have golden tests for all of the layers, but we don't really have much if any at the moment. How do we prioritize this fix, pixel tests for this fix, and pixel tests for the layers in general?

Reply:
We've setup golden testing on the engine side for the web bits. They still rely on the goldens repo today, but we now have [a Skia Gold instance](https://flutter-engine-gold.skia.org/) ready and we plan to move our tests there eventually. Happy to share this with the rest of the engine if that's interesting. However, I'm not sure how much extra value we'd get on top of golden tests that live in the framework repo.

Reply:
@yjbanov : it's great to know that Skia Gold is coming to the engine! Can we also have a wiki page like https://github.com/flutter/flutter/wiki/Writing-a-golden-file-test-for-package%3Aflutter to teach us how to properly write golden tests in the engine repo?

Reply:
@yjbanov what is the schedule for moving to the Skia Gold instance? It might make sense to target that operation for enhancing all of our engine layer pixel testing if "eventually" is more of a scheduling uncertainty than a euphemism for a wish list.

The added value is more targeted testing and testing that happens before we merge to another level in the Flutter forest. Golden testing is still important at the framework level to make sure that the whole project works as a complete package, but goldens in the engine framework make for earlier and more component-wise testing.

Reply:
The engine team has decided to address the lack of pixel verification tests on the layers in the upcoming quarter.

Reply:
Definitely this PR should not wait for Skia Gold integration for the engine. You should be able to write a widget golden test over in the flutter/flutter repo without waiting and get the same signal. We do run framework tests as part of engine presubmit.